Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/TimeslotController.java

```diff
@@ -27,14 +32,24 @@ public class TimeslotController {
     private final TimeslotDtoMapper timeslotDtoMapper;
     private final TimeslotResponseDtoMapper timeslotResponseDtoMapper;
 
-    @PreAuthorize("hasRole('ADMIN')")
+    @PreAuthorize("hasAnyRole('ADMIN','TENNIS_PLAYER')")
     @GetMapping(value = "/")
     public ResponseEntity<List<TimeslotDto>> getAll(){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Person user = (Person)authentication.getPrincipal();
+
         List<TimeslotDto> timeslots = new ArrayList<>();
-        for (Timeslot timeslot: timeslotService.getAll()) {
-            timeslots.add(timeslotDtoMapper.fromTimeslotToTimeslotDto(timeslot));
+        if(user.getRole().getRoleName().equals(UserType.ROLE_TENNIS_PLAYER)) {
+            for (Timeslot timeslot : timeslotService.getAll()) {
```

Comment:
It's better to create a new method in service class to get all timeslots based on person ID (for example **getAllTimeslots(personId)**).